source/fundamentals/crud/read-operations/cursor.txt

```diff
@@ -1,8 +1,266 @@
+.. _cursor-golang:
+
 =========================
 Access Data From a Cursor
 =========================
 
 .. default-domain:: mongodb
 
-Read operations that return multiple documents do not immediately return
-all values matching the query.
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+In this guide, you can learn how to access data with a **cursor**.
+
+A cursor is a mechanism that allows an application to iterate over
+database results while holding a subset of them in memory at a given
+time. The Go driver uses a cursor in some read operations to return
+matched documents in batches as opposed to all at once.
+
+Retrieve Sample Data
+~~~~~~~~~~~~~~~~~~~~
+
+Each section assumes you retrieve all the documents in your collection
+with the ``Find()`` function.
+
+.. literalinclude:: /includes/fundamentals/code-snippets/CRUD/cursor.go
+   :language: go
+   :dedent:
+   :start-after: begin find
+   :end-before: end find
+
+Access Cursor Elements
+----------------------
```

Comment:
Suggestion: As readers must use begin to process their cursor with the content currently located in the "Usage Patterns" section before accessing cursor elements, could this section be moved after the content currently located in the "Usage Patterns" section?